website/docs/guides/container_service_extension.html.markdown

```diff
@@ -0,0 +1,445 @@
+  description          = "ubuntu-2004-kube-v1.21.2+vmware.1-tkg.1-7832907791984498322"
+  ova_path             = "/Users/johndoe/Download/ubuntu-2004-kube-v1.21.2+vmware.1-tkg.1-7832907791984498322.ova"
+  upload_piece_size    = 100
+  show_upload_progress = true
+
+  catalog_item_metadata = {
+    "kind"               = "TKGm"
+    "kubernetes"         = "TKGm"
+    "kubernetes_version" = "v1.21.2+vmware.1"
+    "name"               = "ubuntu-2004-kube-v1.21.2+vmware.1-tkg.1-7832907791984498322"
+    "os"                 = "ubuntu"
+    "revision"           = "1"
+  }
+}
+```
+
+Notice that all the metadata entries from `catalog_item_metadata` are required for CSE to fetch the OVA file:
+* `kind`: Needs to be set to `TKGm` in all cases, as *Native* is not supported yet.
+* `kubernetes`: Same as above.
+* `kubernetes_version`: When the OVA is downloaded from VMware Customer Connect, the version appears as part of the file name.
+* `name`: OVA full file name. VMware Customer Connect should provide already the downloaded OVA with a proper canonical name.
+* `os`: When the OVA is downloaded from VMware Customer Connect, the OS appears as part of the file name.
+* `revision`: Needs to be always `1`. This information is internally used by CSE.
+
+Alternatively, you can upload the OVA file using `cse-cli`. This command line tool is explained in the next step.
+
+### Step 5: CSE cli
+
+This is the only step that can be done without any Terraform script, but you can also use the
+[`null_resource`](https://registry.terraform.io/providers/hashicorp/null/latest/docs/resources/resource) from the
+[null provider](https://registry.terraform.io/providers/hashicorp/null/3.1.1) as shown in the "[Examples](#examples)"
+section below.
+
+In any case, you need to [install CSE command line interface](https://vmware.github.io/container-service-extension/cse3_0/INSTALLATION.html#getting_cse)
+and then provide a YAML configuration file with the entities that were created by Terraform.
+
+You can check the documentation for this configuration file [here](https://vmware.github.io/container-service-extension/cse3_1/CSE_CONFIG.html).
+An example file is provided below, with all the information from the snippets shown in previous steps:
+
+```yaml
+mqtt:
+  verify_ssl: false
+ 
+vcd:
+  host: my-vcd-dev.company.com
+  log: true
+  password: "*****"
+  port: 443
+  username: administrator
+  verify: false
+ 
+service:
+  enforce_authorization: false
+  legacy_mode: false
+  log_wire: false
+  no_vc_communication_mode: true
+  processors: 15
+  telemetry:
+    enable: true
+ 
+broker:
+  catalog: cat-cse
+  ip_allocation_mode: pool
+  network: cse_routed_net
+  org: cse_org
+  remote_template_cookbook_url: https://raw.githubusercontent.com/vmware/container-service-extension-templates/master/template_v2.yaml
+  storage_profile: '*'
+  vdc: cse_vdc
+```
+
+When you execute the following command:
+
+```shell
+cse install -c config.yaml
```

Comment:
This must be run from the CSE machine, so this will also not be done by Terraform?

Reply:
We managed to run it inside Terraform through the `null_resource`. This is done in the examples section (link above) while here we explain the "manual" way of doing it.

@lvirbalas @Didainius @vbauzysvmware Maybe it's a good idea to remove this and just put the `null_resource` way of achieving this?